packages/core/src/components/widgets/SelectWidget.tsx

```diff
@@ -67,7 +67,37 @@ function SelectWidget<T = any, S extends StrictRJSFSchema = RJSFSchema, F extend
   );
 
   const selectedIndexes = enumOptionsIndexForValue<S>(value, enumOptions, multiple);
-  const showPlaceholderOption = !multiple && schema.default === undefined;
+
+  let opts = null
+  if (optgroups) {
+    let enumOptionFromValue = new Map()
+    enumOptions.forEach((enumOption, i) => {
+      enumOptionFromValue.set(enumOption.value, [enumOption, i])
+    })
+    opts = Object.keys(optgroups).map(optgroup => {
+      return (
+        <optgroup key={optgroup} label={optgroup}>{optgroups[optgroup].map(value => {
+          if (!enumOptionFromValue.has(value)) return null
+          const disabled = enumDisabled && enumDisabled.indexOf(value) !== -1;
+          let [{ label }, i] = enumOptionFromValue.get(value)
+          return (
+            <option key={i} value={String(i)} disabled={disabled}>
+              {label}
+            </option>
+          );
```

Comment:
Duplicate code here, minus the `let` line, which really should be const since you don't change anything

Reply:
Using the suggestion below, this would look like:
```
          const [{ label }, i] = enumOptionFromValue.get(value)
          return <Option key={i} enumDisabled={enumDisabled} index={i} value={value} label={label} />;
```